make/tools.mk

```diff
@@ -193,13 +192,23 @@ EXE_FLAGS          := $(EXE_FLAGS_EXT) $(FLAG_RELRO) -Wl,--gc-sections
 SO_FLAGS           := $(SO_FLAGS_EXT) $(FLAG_RELRO) -Wl,--gc-sections -shared $(FLAG_STDLIB) -fPIC 
 
 # Define flags for host build
-HOST_CFLAGS        := $(CFLAGS)
-HOST_CDEFS         := $(CDEFS)
-HOST_CXXFLAGS      := $(CXXFLAGS)
-HOST_CXXDEFS       := $(CXXDEFS)
-HOST_LDFLAGS       := $(LDFLAGS)
-HOST_EXE_FLAGS     := $(EXE_FLAGS)
-HOST_SO_FLAGS      := $(SO_FLAGS)
+ifeq ($(CROSS_COMPILE),1)
```

Comment:
I think this still looks not very well. I think we can improve all this stuff, too. I don't like the `filter-out` magic, we can compute flags as it done for `ARCHITECTURE` and `HOST_ARCHITECTURE` variables.